packages/ebsi-authorization-client/src/agent/EBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,363 @@
+import { CredentialPayload, IAgentPlugin } from '@veramo/core'
+import { GetOIDProviderMetadataSuccessResponse, IRequiredContext, OpenIDScope, schema, ScopeByDefinition } from '../index'
+import {
+  CreateOAuth2SessionArgs,
+  CreateOAuth2SessionResponse,
+  CreateSIOPSessionArgs,
+  CreateSIOPSessionResponse,
+  ExceptionResponse,
+  GetAccessTokenArgs,
+  GetAccessTokenResponse,
+  GetOIDProviderJwksResponse,
+  GetOIDProviderMetadataResponse,
+  GetPresentationDefinitionArgs,
+  GetPresentationDefinitionResponse,
+  IEBSIAuthorizationClient,
+  InitiateSIOPDidAuthRequestArgs,
+  InitiateSIOPDidAuthRequestResponse,
+} from '../types/IEBSIAuthorizationClient'
+import { uuid } from 'uuidv4'
+import { Descriptor } from '@sphereon/pex-models'
+import { JWTHeader, JWTPayload } from 'did-jwt'
+import { exportJWK, importJWK, JWK, SignJWT } from 'jose'
+import * as u8a from 'uint8arrays'
+import { JsonWebKey, JsonWebKey2020 } from '@transmute/json-web-signature'
+
+export class EBSIAuthorizationClient implements IAgentPlugin {
+  readonly schema = schema.IEBSIAuthorizationClient
+  readonly methods: IEBSIAuthorizationClient = {
+    getOIDProviderMetadata: this.getOIDProviderMetadata.bind(this),
+    getOIDProviderJwks: this.getOIDProviderJwks.bind(this),
+    getPresentationDefinition: this.getPresentationDefinition.bind(this),
+    getAccessToken: this.getAccessToken.bind(this),
+    initiateSIOPDidAuthRequest: this.initiateSIOPDidAuthRequest.bind(this),
+    createSIOPSession: this.createSIOPSession.bind(this),
+    createOAuth2Session: this.createOAuth2Session.bind(this),
+  }
+  private configuration: GetOIDProviderMetadataSuccessResponse
+
+  //TODO Pass the configuration manually or call init
+  constructor(readonly baseUrl: string = 'https://api-pilot.ebsi.eu/authorisation/v4/') {
+    // TODO call configuration endpoint
```

Comment:
Not from constructor. Rather would see a lazy method being called as a first action of any method that needs the metdata